modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js

```diff
@@ -81,13 +99,78 @@ export default function AuditBarChart({rtl, vocabularies}) {
 		? [BAR_CHART.dotRadiusMin]
 		: [BAR_CHART.legendHeight + BAR_CHART.dotRadiusMin];
 
+	const [checkboxes, setCheckbox] = useState(legendCheckboxes);
+
+	useEffect(() => {
+		if (Object.keys(colors).length) {
+			const head =
+				document.head || document.getElementsByTagName('head')[0];
+			const style = document.createElement('style');
+			style.id = 'legend-checkboxes';
+			style.type = 'text/css';
+
+			{
+				Object.entries(colors).forEach(([dataKey, color]) => {
+					style.textContent += `.custom-control-color-${dataKey}.custom-control-input:checked ~ 
+					.custom-control-label::before {
+						background-color: ${color};
+						border-color: ${color};
+				}
+				.custom-control-color-${dataKey}.custom-control-input:not(:checked) ~ 
+					.custom-control-label::before {
+						border-color: ${color};
+				}`;
+				});
+			}
+
+			head.appendChild(style);
+		}
+	}, [colors]);
+
+	useEffect(() => {
+		return () => {
```

Comment:
Why don't you return this from the previous `useEffect`? This feels like that one's cleanup... 🤔

Reply:
Also the fact that the other one has `[colors]` as dependencies and this one has none makes we wonder...

Reply:
Done 👍